server/test/requestRoutes.spec.js

```diff
@@ -0,0 +1,167 @@
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+const app = require("../app.js");
+const { Profile, User } = require("../models");
+const mongoose = require("mongoose");
+
+const { expect } = chai;
+
+chai.should();
+chai.use(chaiHttp);
+
+const testingIds = [];
+let agent;
+
+describe("Request Routes", () => {
+  before(done => {
+    mongoose.connect("mongodb://localhost/team-pine-cone", {
+      useNewUrlParser: true,
+      useUnifiedTopology: true,
+      useCreateIndex: true,
+    });
+    const dbc = mongoose.connection;
+    agent = chai.request.agent(app);
+    dbc.on("error", console.error.bind(console, "connection error"));
+    dbc.once("open", function () {
+      const testProfile = new Profile({
+        firstName: "Joe",
+        lastName: "Smith",
+        birthDate: new Date("01-01-1990"),
+        phone: "+1 (555) 555-5555",
+        location: {
+          type: "Point",
+          coordinates: [-122.5, 37.7],
+        },
+      });
+      testProfile.save().then(profile => {
+        testingIds.push(profile._id);
+        const testProfile2 = new Profile({
+          firstName: "Jim",
+          lastName: "Smith",
+          birthDate: new Date("01-01-1990"),
+          phone: "+1 (555) 555-5555",
+          location: {
+            type: "Point",
+            coordinates: [-122.5, 37.7],
+          },
+        });
+        testProfile2.save().then(profile => {
+          testingIds.push(profile._id);
+          const test = {
+            email: "joe@test.com",
+            password: "joetest"
+          };
+          const testUser = new User({
+            ...test,
+            profile: testingIds[0]
+          });
+          testUser.save().then(() => {
+            agent
+              .post("/login")
+              .send({
+                email: test.email,
+                password: test.password
+              })
+              .end(() => {
+                done();
+              });
+          });
+        });
+      });
+    });
+  });
+  let requestId;
+  it("POST /request/create Creates a request with a correct POST request", done => {
```

Comment:
might be good to update this test with the user_id somehow and make sure it's symmetric?